src/libraries/Common/src/System/Security/Cryptography/MLDsa.cs

```diff
@@ -507,6 +494,223 @@ public bool VerifyPreHash(byte[] hash, byte[] signature, string hashAlgorithmOid
+
+            return SignExternalMu(new ReadOnlySpan<byte>(mu));
+        }
+
+        /// <summary>
+        ///   Signs the specified externally computed signature mu (&#x3BC;) value.
+        /// </summary>
+        /// <param name="mu">
+        ///   The signature mu value to sign.
+        /// </param>
+        /// <returns>
+        ///   ML-DSA signature for the specified mu value.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///   The buffer in <paramref name="mu"/> is the incorrect length for the signature mu value.
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">
+        ///   This instance has been disposed.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <para>The instance represents only a public key.</para>
+        ///   <para>-or-</para>
+        ///   <para>An error occurred while signing the hash.</para>
+        /// </exception>
+        /// <seealso cref="OpenExternalMuHash(byte[])"/>
+        /// <seealso cref="VerifyExternalMu(byte[], byte[])"/>
+        public byte[] SignExternalMu(ReadOnlySpan<byte> mu)
+        {
+            byte[] destination = new byte[Algorithm.SignatureSizeInBytes];
+            SignExternalMu(mu, destination.AsSpan());
+            return destination;
+        }
+
+        /// <summary>
+        ///   Signs the specified externally computed signature mu (&#x3BC;) value,
+        ///   writing the signature into the provided buffer.
+        /// </summary>
+        /// <param name="mu">
+        ///   The signature mu value to sign.
+        /// </param>
+        /// <param name="destination">
+        ///   The buffer to receive the signature. Its length must be exactly
+        ///   <see cref="MLDsaAlgorithm.SignatureSizeInBytes"/>.
+        /// </param>
+        /// <exception cref="ArgumentException">
+        ///   <para>
+        ///     The buffer in <paramref name="mu"/> is the incorrect length for the signature mu value.
+        ///   </para>
+        ///   <para>-or-</para>
+        ///   <para>
+        ///     The buffer in <paramref name="destination"/> is the incorrect length to receive the signature.
+        ///   </para>
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">
+        ///   This instance has been disposed.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <para>The instance represents only a public key.</para>
+        ///   <para>-or-</para>
+        ///   <para>An error occurred while signing the hash.</para>
+        /// </exception>
+        /// <seealso cref="OpenExternalMuHash(ReadOnlySpan{byte})"/>
+        /// <seealso cref="VerifyExternalMu(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
+        public void SignExternalMu(ReadOnlySpan<byte> mu, Span<byte> destination)
+        {
+            Helpers.ThrowIfWrongLength(mu, Algorithm.MuSizeInBytes);
```

Comment:
I don't think this helper is appropriate here because the exception message is "Destination must be exactly {0} bytes.". mu is not a destination.